platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -1030,3 +1030,165 @@ def get_sfp(self, index):
+                from .module import DpuModule
+                self._module_list[index] = DpuModule(index + 1)
+                self.module_name_index_map[self._module_list[index].get_name()] = index
+                self.module_initialized_count += 1
+
+    def initialize_modules(self):
+        count = self.get_num_modules()
+        for index in range(count):
+            self.initialize_single_module(index=index)
+
+    def get_num_modules(self):
+        """
+        Retrieves the number of modules available on this chassis
+        On a SmarSwitch chassis this includes the number of DPUs.
+
+        Returns:
+            An integer, the number of modules available on this chassis
+        """
+        return DeviceDataManager.get_dpu_count()
+
+    def get_all_modules(self):
+        """
+        Retrieves all modules available on this chassis. On a SmarSwitch
+        chassis this includes the number of DPUs.
+
+        Returns:
+            A list of objects derived from ModuleBase representing all
+            modules available on this chassis
+        """
+        self.initialize_modules()
+        return self._module_list
+
+    def get_module(self, index):
+        """
+        Retrieves module represented by (0-based) index <index>
+        On a SmartSwitch index:0 will fetch switch, index:1 will fetch
+        DPU0 and so on
+
+        Args:
+            index: An integer, the index (0-based) of the module to
+            retrieve
+
+        Returns:
+            An object dervied from ModuleBase representing the specified
+            module
+        """
+        self.initialize_single_module(index)
+        return super(SmartSwitchChassis, self).get_module(index)
+
+    def get_module_index(self, module_name):
+        """
+        Retrieves module index from the module name
+
+        Args:
+            module_name: A string, prefixed by SUPERVISOR, LINE-CARD or FABRIC-CARD
+            Ex. SUPERVISOR0, LINE-CARD1, FABRIC-CARD5
+            SmartSwitch Example: SWITCH, DPU1, DPU2 ... DPUX
+
+        Returns:
+            An integer, the index of the ModuleBase object in the module_list
+        """
+        # TODO: Confirm whether Switch is module or not
+        ret_val = -1
+        try:
+            ret_val = self.module_name_index_map[module_name]
+        except KeyError as e:
+            logger.log_error("Failed to obtain name to index mapping! Check module initialization {}".format(repr(e)))
+        return ret_val
+
+    ##############################################
+    # SmartSwitch methods
+    ##############################################
+
+    def get_dpu_id(self, name):
+        """
+        Retrieves the DPU ID for the given dpu-module name.
+        Returns None for non-smartswitch chassis.
+        Returns:
+            An integer, indicating the DPU ID Ex: name:DPU0 return value 1,
+            name:DPU1 return value 2, name:DPUX return value X+1
+        """
+        return self.get_module_index(name)
```

Comment:
From the decription the dpu ID is module ID + 1. Isn't it?

Reply:
I thought that this was because the index 0 indicates the `switch` module (which we are not considering, which is the reason the first DPU (DPU0) returns value 1), should I align it with the description?

Reply:
After discussion, changing  the return value to match the name, We have dpu1, dpu2, dpu3, dpu4 and the return values would be 1, 2, 3, 4

Reply:
Fixed